components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credentials-list/node-credential-list.component.ts

```diff
@@ -0,0 +1,144 @@
+  public nodeCredentialToDelete: NodeCredential;
+  public createModalVisible = false;
+  public creatingNodeCredential = false;
+  public deleteModalVisible = false;
+  public nodesListLoading = true;
+  public messageModalVisible = false;
+  public nodesListSort = 'last_contact';
+  public nodesListSortOrder = 'DESC';
+  public nodesListFilters = [{key: 'type', values: ['ssh', 'winrm', 'sudo']}];
+  public sortBy: string;
+  public params: object;
+
+  constructor(
+    private store: Store<NgrxStateAtom>,
+    private layoutFacade: LayoutFacadeService
+  ) {
+    this.loading$ = store.select(getAllStatus).pipe(map(loading));
+  }
+
+  ngOnInit(): void {
+    this.layoutFacade.showSidebar(Sidebar.Settings);
+    this.getNodeList();
+
+  }
+
+  ngOnDestroy() {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  public openCreateModal(): void {
+    this.openUserModal.emit(true);
+  }
+
+  orderFor(column) {
+    return column === this.sortBy ? this.orderBy : 'none';
+  }
+
+  handleSortToggle({detail: sortParams}) {
+
+    this.store.dispatch(new SortNodeCredentialList(sortParams));
+    this.getNodeList();
+  }
+
+  formatKeyType(credentialType: NodeCredentialTypes): string {
+    switch (credentialType) {
+      case NodeCredentialTypes.SSH:
+        return 'SSH';
+      case NodeCredentialTypes.WinRM:
+        return 'WinRM';
+      case NodeCredentialTypes.Sudo:
+        return 'Sudo';
+      default:
+        return 'SSH';
+    }
+  }
+
+  public startNodeCredentialDelete( $event: MatOptionSelectionChange, p: NodeCredential): void {
+    if ($event.isUserInput) {
+      this.nodeCredentialToDelete = p;
+      this.deleteModalVisible = true;
+    }
+  }
+
+  getNodeList() {
+    this.instanceNodeCredentials$ = this.store.pipe(select(allCredentials));
+    this.store.select(nodeCredentialListState).subscribe((state) => {
+      this.params = pick(['filters', 'page', 'per_page'], state);
+
+      this.sortBy = get('sort', state);
+      this.orderBy = get('order', state);
+
+      if (this.orderBy !== 'none') {
+        this.params = pipe(
+          set('order', toUpper(this.orderBy)),
+          set('sort', this.sortBy)
+        )(this.params) as object;
+      }
+    });
+    this.store.dispatch(new NodeCredentialsSearch(this.params));
+    this.nodesListLoading = false;
+  }
+
+  public deleteNodeCredential(): void {
+    this.closeDeleteModal();
+    this.store.dispatch(new DeleteNodeCredential(this.nodeCredentialToDelete));
+  }
+
+  public closeDeleteModal(): void {
+    this.deleteModalVisible = false;
+  }
+
+
+  public closeMessageModal(): void {
+    this.messageModalVisible = false;
+  }
+
+
+  public getLastModified(lastModifiedDate): string {
+    return moment.utc(lastModifiedDate).format(DateTime.RFC2822);
+  }
+
+  public inUseMessage(): string {
+    return (this.nodeCredentialToDelete && this.nodeCredentialToDelete.data.length > 0) ?
+      'and will disrupt access for all attached members' : '';
```

Comment:
This wording (copied from PolicyListComponent) is not appropriate here; policies have members but node credentials do not. What are the data associated with a node credential?